docs/tutorials/chatbot.md

```diff
@@ -0,0 +1,571 @@
+
+### Create a Flow
+
+The very first concept you'll see in Jina is a Flow. You can see {ref}`here <flow-cookbook>` a more formal introduction of what it is, but for now, think of the Flow as a manager in Jina, it takes care of the all the tasks that will run on your application and each Flow object will take care of one real-world task.
+
+To create a Flow you only need to import it from Jina. So open your favorite IDE, create an `app.py` file and let's start writing our code:
+
+``` python
+from jina import Flow
+flow = Flow()
+```
+
+But this is an empty Flow. Since we want to encode our data and then index it, we need to add elements to it. The only things we add to a Flow are Executors. We will talk about them more formally later, but think of them as the elements that will do all the data processing you want.
+
+### Add elements to a Flow
+
+To add elements to your Flow you just need to use the `.add()` method. You can add as many pods as you wish.
+
+``` python
+from jina import Flow
+flow = Flow().add().add()
+```
+
+And for our example, we need to add two `Executors`:
+
+1.  A transformer (to encode our data)
+2.  An indexer
+
+So add the following to our code:
+
+``` python
+from jina import Flow
+flow = (
+        Flow()
+        .add(uses=MyTransformer)
+        .add(uses=MyIndexer)
+    )
+```
+
+Right now we haven't defined `MyTransformer` or `MyIndexer`. Let's create some dummy Executors so we can try our app. These will not be our final Executors but just something basic to learn first.
+
+### Create dummy Executors
+
+Now we have a Flow with two Executors. Write the following in your code:
+
+``` python
+from jina import Executor, requests
+class MyTransformer(Executor):
+    @requests(on='/foo')
+    def foo(self, **kwargs):
+        print(f'foo is doing cool stuff: {kwargs}')
+class MyIndexer(Executor):
+    @requests(on='/bar')
+    def bar(self, **kwargs):
+        print(f'bar is doing cool stuff: {kwargs}')
+```
+
+We will have more complex Executors later. For now our two Executors are only printing a line.
+
+So far it's a simple Flow but it is still useful to visualize it to make sure we have what we want.
+
+### Visualize a Flow
+
+By now, your code should look like this:
+
+``` python
+from jina import Flow, Document, Executor, requests
+class MyTransformer(Executor):
+    @requests(on='/foo')
+    def foo(self, **kwargs):
+        print(f'foo is doing cool stuff: {kwargs}')
+class MyIndexer(Executor):
+    @requests(on='/bar')
+    def bar(self, **kwargs):
+        print(f'bar is doing cool stuff: {kwargs}')
+flow = (
+        Flow()
+        .add(uses=MyTransformer)
+        .add(uses=MyIndexer)
+    )
+```
+
+If you want to visualize your Flow you can do that with `plot`. So add the `.plot()` function at the end of your Flow
+
+``` python
+from jina import Flow
+flow = (
+        Flow()
+        .add(uses=MyTransformer)
+        .add(uses=MyIndexer)
+        .plot('our_flow.svg')
```

Comment:
I would prefer fist defining the flow, and then calling `flow.plot()`

Reply:
done